cdap-tms/src/main/java/co/cask/cdap/messaging/service/CoreMessagingService.java

```diff
@@ -247,6 +259,29 @@ protected void startUp() throws Exception {
     LOG.info("Core Messaging Service started");
   }
 
+  /**
+   * Creates the given topic if it is not yet created. Adds a topic to the asyncCreationTopics if creation fails.
+   */
+  private void createSystemTopic(String topicName,  Queue<TopicId> asyncCreationTopics) {
```

Comment:
It is fine to call `asyncCreationTopics` outside of this method, as it is indeed queuing up for some async operation. However, for this method, there is no such knowledge. So for this method, it's better to call it `createFailureTopics`